### DISCUSSION TITLE
OpenAI o1 is now available in GitHub Copilot Chat in public preview

### DISCUSSION THREAD

**Discussion by Akash1134** [community/community#142851] [Copilot News and Announcements]
![unnamed](https://github.com/user-attachments/assets/66bbbeac-b07b-4912-ad07-fc6ad33c2a49)

👀 **Big news**—we’re opening up access to OpenAI’s o1-preview and o1-mini models for everyone in [GitHub Copilot](https://github.com/copilot). You can now test them out in VS Code, Visual Studio, GitHub.com, and the GitHub Models playground. 🎉

So, what’s the story with the o1-preview and o1-mini models ? Think of these models as your ultra-smart coding sidekicks—they thrive on tackling complex problems, hunting down tricky edge cases, and suggesting thoughtful solutions. Basically, it’s like having an AI buddy that just clicks with your code.

Starting today, you get to play around with both models and switch between them on the fly! Whether you’re breaking down an API, whipping up some code snippets, or untangling a gnarly bug, pick the model that fits the moment and let them bring your ideas to life. 🧩✨ 

<img width="50%" alt="Screenshot 2024-10-29 at 17 03 12" src="https://github.com/user-attachments/assets/31c3ed3c-1057-41ae-9ec3-f8b36185cd25" >

But hey, we want to hear from you—jump in, test them out, and voice your feedback! What’s working, what’s surprising, and where can we make things even better? Let’s build something amazing together. 🚀 

Explore [how we’re bringing developer choice to copilot](https://github.blog/news-insights/product-news/bringing-developer-choice-to-copilot/) with anthropic’s Claude 3.5 sonnet, Google’s Gemini 1.5 pro, and OpenaAI’s O1-preview.

**Comment by armesha** [comment]
Are there any daily limits for the o1-preview and o1-mini models?

**Reply by dstampher** [reply]
I haven't been able to find this information anywhere :-(

Hopefully we can find an answer soon.

**Reply by GitMurf** [reply]
I started getting an error 429 after using o1-preview for not very long... maybe 10 messages today. In googling it, 429 appears to be a rate limiting error so seems like there is some sort of daily limit and unfortunately seems pretty small :-(

**Reply by dstampher** [reply]
> I started getting an error 429 after using o1-preview for not very long... maybe 10 messages today. In googling it, 429 appears to be a rate limiting error so seems like there is some sort of daily limit and unfortunately seems pretty small :-(

The current limit for ChatGPT Plus users is 50 per week for o1-preview, and 50 per day for o1-mini.

So, even if it is 10 per day with o1-preview, that would still be significantly more per month than even a paid ChatGPT sub.

EDIT:  After some digging, I seem to have finally found some usage limits, but they are for the Github Models playground.  However, based on the fact that you said you got 429 errors after ~10 o1-preview messages, it seems that these usage limits might also apply to Copilot:  https://docs.github.com/en/github-models/prototyping-with-ai-models#rate-limits

In which case the limit would be 8 per day with o1-preview, and 12 per day with o1-mini.

**Comment by sheldonhull** [comment]
Does this require accepting preview of models?
My org won't accept preview/early access, so not sure if I should be able to still use o1-preview without opting into the preview of models for this.
I have azure openai o1-preview as well, but I haven't seen any mention I can track down of being able to connect my own enterprise deployment to it.

**Reply by samus-aran** [reply]
Hi @sheldonhull 👋

For both o1s your organisation would need to accept the preview access. If you have copilot access via your enterprise/org and the enterprise/org has disabled it, you would not be able to use it.

**Comment by kangkannnng** [comment]
Can we use api to get access?

**Reply by samus-aran** [reply]
Hi @kangkannnng, unfortunately not. API usage of these are only currently available with our integration partners. Unauthorised use will incur to warnings and suspensions.
Alternatively you can use the API via agents! Here are the docs: https://docs.github.com/en/copilot/building-copilot-extensions/building-a-copilot-agent-for-your-copilot-extension/using-copilots-llm-for-your-agent. Additionally, you are able to experiment using GitHub Models which does have o1 available: https://github.com/marketplace/models 👍

**Comment by morty6688** [comment]
When can we use o1 in jetbrains IDE?

**Reply by Akash1134** [reply]
The OpenAI O1 models are currently available as public preview in with VS Code, Visual Studio, GitHub.com, and the GitHub Models playground. 

Thanks!

**Reply by joshualyon** [reply]
Hey @Akash1134 thanks for sharing which IDEs and locations o1 is *currently* available. What about the question @morty6688 asked though - what's the plan for introducing these models to JetBrains IDEs? 

The current GitHub Copilot plugin for Jetbrains IDEs does *not* support the preview o1 or Claude Sonnet choices as of the time this comment was written.

**Reply by SirusDoma** [reply]
I just cancelled my newly subscribed copilot due to this. I primarily use JB IDE and based on what I read, the extension is lagging behind with vscode extension. This post is the proof that I should avoid github copilot and use other solution instead.

**Reply by morty6688** [reply]
@Akash1134 Hi, it's been almost two months, any progress?

**Comment by greg34x** [comment]
@Akash1134 
What are o1 preview and o1 mini limits on `Visual Studio 2022`?
![image](https://github.com/user-attachments/assets/8f8cceae-3601-462b-91d9-781a359d67a5)
Also, why Claude is missing on VS22 Copilot Chat?

**Reply by Akash1134** [reply]
Hi @greg34x , Currently, I don't have the answer for you but I will update as soon as I have one. Thanks! 🙂

**Reply by greg34x** [reply]
@Akash1134  so? we would like to use Claude on Visual Studio, we dont use VSCode.

**Comment by simlei** [comment]
When is the anthropic + general model selection coming to copilot.vim? Not for long and I'm ditching my subscription if there's this kind of exclusivity without any reason but lock-in design. There's plenty workable alternatives anyway my guys.

**Comment by LeoX91** [comment]
Hello, I'm with enterprise licence for Copilot on Visual Studio Code. The problem here is that the model selector only shows Claude Sonnet and Gpt4o, nothing regarding o1 even if my SysAdmin enabled them :(

Any hint?

**Reply by kmrgirish** [reply]
same for me, o1 models are missing from vscode selection although o1 models are enabled by my organisation. When selecting the o1-mini model from [zed](https://zed.dev/), I am receiving bellow 400 bad req. 

<img width="395" alt="image" src="https://github.com/user-attachments/assets/05ea9c8a-cc0b-45ad-adc3-d32bf11bfc52">

**Reply by jpierzchala** [reply]
Same for me as well

**Comment by AstrafoxMariuszPiotrzkowski** [comment]
### How do I switch model in Visual Studio 2022? I have everything enabled (It's managed by an organization):
![obraz](https://github.com/user-attachments/assets/17eb9ca0-0482-4d2a-b04f-3117a6e6fd35)
### But I see no way to change the model:
![obraz](https://github.com/user-attachments/assets/81ee05d7-170f-4d90-8b9f-379099cb534a)
I've tried waiting (1 day), logging out and back in and starting new conversations.

**Reply by greg34x** [reply]
I'm not sure but i **think**, that at the moment its available only on **Visual Studio Preview**.

**Reply by NikhilSharma040997** [reply]
You can download the VS Preview version and then you will be able to see the options for models in the visual studio you are using.

**Comment by rkaiserhuyett** [comment]
When we started using GitHub Copilot in VS, we were informed that the code we entered would be secured, not used for model training and not available to the world. If we switch models, is it still secured, not used for training and not available to the world? We have looked for verification of this and have not found anything concrete.
